src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -201,6 +204,25 @@ private static String parseGroupNameFromConfigurationArn(Configuration config) {
         return groupName;
     }
 
+    private static String parseGroupNameFromIotArn(String arn) {
+        if (arn == null || arn.isEmpty()) {
+            return null;
+        }
+
+        String groupName;
+        try {
+            // IoTJob Arn only gives thing group name without 'thinggroup/' prefix
+            groupName = Arn.fromString(arn).resource().resource();
+            if (groupName != null) {
+                groupName = new StringBuilder(THING_GROUP_RESOURCE_NAME_PREFIX).append(groupName).toString();
+            }
+        } catch (IllegalArgumentException e) {
+            // so that it can proceed, rather than fail, when the format of configurationArn is wrong.
```

Comment:
This is similar to configuration arn behavior. I was expecting this to be already handled, as the same thing can happen with a bad configuration arn.

Could you shed some light on why this would be any different?

Reply:
It may have been a miss for existing configuration handling or maybe it didn't matter earlier since there was only one arn and as long as that was same across deployments, dependency resolution would work fine. Now there's an additional arn which has to be same as the group arn it's mimicking, if one is malformed/wrong then there will be a conflict - so you don't get the `onBehalfOf` behavior you expect, and customers won't know why there's a version conflict.

Reply:
Adding explicit exception then:

`throw new InvalidRequestException("Malformed value in deployment configuration for 'onBehalfOf'.");`